janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/compat/AbstractESCompat.java

```diff
@@ -53,6 +56,17 @@
         ;
     }
 
+    public static AbstractESCompat acquireCompatForVersion(ElasticMajorVersion elasticMajorVersion) throws BackendException {
```

Comment:
I am not sure this is a good idea to place this logic here. The logic looks like either Adapter or Builder pattern.
ES6Compat and ES7Compat are children of AbstractESCompat. By extending it they will also be able to use this method. I.e. ES6Compat may have a possibility to create ES7Compat and vice versa which doesn't make sense. IMHO it would be better to move this logic somewhere else

Reply:
Added factory class, should be better now.